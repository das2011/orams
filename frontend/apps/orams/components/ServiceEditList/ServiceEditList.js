import React from 'react'
import { uniqueID } from 'shared/utils/helpers'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import styles from './ServiceEditList.scss'

const ServiceEditList = props => {
  const { editServiceData, linkClick, successMessage, hideNav, supplierCode, supplier, isCeilingPriceUpdate } = props
  const { services } = editServiceData

  const renderSuccessMessage = () => {
    const message = isCeilingPriceUpdate ? 'Ceiling price updated' : 'Pricing updated'

    if (successMessage) {
      return (
        <AUpageAlert as="success">
          <h4>
            {message}
          </h4>
        </AUpageAlert>
      )
    }

    return ''
  }

  const renderHeading = () => {
    if (isCeilingPriceUpdate) {
      const supplierName = supplier ? supplier.name : ''

      return (
        <header>
          <h1 className="au-display-xl">
            {supplierName}
          </h1>
          <h2 className="au-display-lg">Update ceiling prices - select a service</h2>
          <div className={styles.stepTitle}>Step 1 of 3</div>
          <div
            role="button"
            tabIndex="0"
            className={`${styles.link} link-back`}
            onClick={() => {
              props.history.goBack()
            }}
          >
            Back to suppliers
          </div>
          <div className={styles.heading}>Select the service you want to edit</div>
        </header>
      )
    }

    return (
      <header>
        <h1 className="au-display-xl">Pricing</h1>
        <div className={styles.stepTitle}>Step 1 of 4</div>
        <div className={styles.heading}>Select the service you want to edit</div>
      </header>
    )
  }

  return (
    <div className={styles.container}>
      {renderSuccessMessage()}
      {renderHeading()}
      <article role="main">
        {services &&
          services.map((service, id = uniqueID()) =>
            <div key={id}>
              {service.subCategories.map((subCategory, subId = uniqueID()) =>
                <div
                  key={subId}
                  onClick={() => {
                    hideNav(true)
                    linkClick(
                      supplierCode,
                      service.id,
                      subCategory.id,
                      service.name,
                      subCategory.name ? subCategory.name : ''
                    )
                    window.scrollTo(0, 0)
                  }}
                >
                  <div className={`${styles.link} link-service`}>
                    {service.name}
                    <span>
                      {subCategory.name ? ` (${subCategory.name})` : ''}
                    </span>
                  </div>
                </div>
              )}
            </div>
          )}
      </article>
    </div>
  )
}

export default ServiceEditList
