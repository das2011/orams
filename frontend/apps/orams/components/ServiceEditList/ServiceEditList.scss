.container {
  margin-bottom: 100px;
}

.link {
  font-size: 17px;
  font-weight: bold;
  color: #202751;
  margin-top: 27px;
  text-decoration: underline;
  cursor: pointer;
}

.stepTitle {
  font-size: 20px;
  color: #777777;
}

.heading {
  font-size: 24px;
  font-weight: bold;
  color: #313131;
  margin-top: 39px;
}
