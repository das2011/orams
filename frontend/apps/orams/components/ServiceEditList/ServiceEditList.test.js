import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import expect from 'expect'
import snapshotRenderer from 'react-test-renderer'
import ServiceEditList from './ServiceEditList'

Enzyme.configure({ adapter: new Adapter() })

describe('#ServiceEditList', () => {
  const goBackSpy = jest.fn()
  const defaultProps = {
    history: {
      goBack: goBackSpy
    },
    hideNav: jest.fn(),
    supplierCode: 1,
    supplier: {
      name: 'Supplier Name'
    },
    editServiceData: {
      services: [
        {
          id: 1,
          name: 'Componensation Service',
          subCategories: [
            {
              id: 1,
              name: 'Physical'
            }
          ]
        },
        {
          id: 2,
          name: 'Rehabiliation Service',
          subCategories: [
            {
              id: 1,
              name: 'Physical'
            },
            {
              id: 1,
              name: 'Psychological'
            }
          ]
        }
      ]
    }
  }

  test('renders correct header when it is update ceiling price', () => {
    const isCeilingPriceUpdate = true
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate,
      linkClick: jest.fn()
    }
    const tree = snapshotRenderer.create(<ServiceEditList {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correct header when it is not update ceiling price', () => {
    const isCeilingPriceUpdate = false
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate,
      linkClick: jest.fn()
    }
    const tree = snapshotRenderer.create(<ServiceEditList {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders success message', () => {
    const isCeilingPriceUpdate = true
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate,
      successMessage: 'Success',
      linkClick: jest.fn()
    }
    const tree = snapshotRenderer.create(<ServiceEditList {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('link is clicked with correct params when selecting a service to edit', () => {
    const isCeilingPriceUpdate = true
    const linkClickSpy = jest.fn()
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate,
      linkClick: linkClickSpy
    }
    const { supplierCode, editServiceData: { services } } = props
    const [compensationService] = services
    const { id: serviceId, name: serviceName, subCategories: [phsysicalCategory] } = compensationService

    const wrapper = mount(<ServiceEditList {...props} />)

    const serviceLink = wrapper.find('.link-service').first()
    serviceLink.simulate('click')

    expect(linkClickSpy).toBeCalledWith(
      supplierCode,
      serviceId,
      phsysicalCategory.id,
      serviceName,
      phsysicalCategory.name
    )
  })
})
