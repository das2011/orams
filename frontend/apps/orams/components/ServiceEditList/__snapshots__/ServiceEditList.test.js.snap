// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ServiceEditList renders correct header when it is not update ceiling price 1`] = `
<div
  className="container"
>
  
  <header>
    <h1
      className="au-display-xl"
    >
      Pricing
    </h1>
    <div
      className="stepTitle"
    >
      Step 1 of 4
    </div>
    <div
      className="heading"
    >
      Select the service you want to edit
    </div>
  </header>
  <article
    role="main"
  >
    <div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Componensation Service
          <span>
             (Physical)
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Rehabiliation Service
          <span>
             (Physical)
          </span>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Rehabiliation Service
          <span>
             (Psychological)
          </span>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`#ServiceEditList renders correct header when it is update ceiling price 1`] = `
<div
  className="container"
>
  
  <header>
    <h1
      className="au-display-xl"
    >
      Supplier Name
    </h1>
    <h2
      className="au-display-lg"
    >
      Update ceiling prices - select a service
    </h2>
    <div
      className="stepTitle"
    >
      Step 1 of 3
    </div>
    <div
      className="link link-back"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Back to suppliers
    </div>
    <div
      className="heading"
    >
      Select the service you want to edit
    </div>
  </header>
  <article
    role="main"
  >
    <div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Componensation Service
          <span>
             (Physical)
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Rehabiliation Service
          <span>
             (Physical)
          </span>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Rehabiliation Service
          <span>
             (Psychological)
          </span>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`#ServiceEditList renders success message 1`] = `
<div
  className="container"
>
  <div
    className="au-body au-page-alerts   au-page-alerts--success"
    role="alert"
  >
    <h4>
      Ceiling price updated
    </h4>
  </div>
  <header>
    <h1
      className="au-display-xl"
    >
      Supplier Name
    </h1>
    <h2
      className="au-display-lg"
    >
      Update ceiling prices - select a service
    </h2>
    <div
      className="stepTitle"
    >
      Step 1 of 3
    </div>
    <div
      className="link link-back"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Back to suppliers
    </div>
    <div
      className="heading"
    >
      Select the service you want to edit
    </div>
  </header>
  <article
    role="main"
  >
    <div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Componensation Service
          <span>
             (Physical)
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Rehabiliation Service
          <span>
             (Physical)
          </span>
        </div>
      </div>
      <div
        onClick={[Function]}
      >
        <div
          className="link link-service"
        >
          Rehabiliation Service
          <span>
             (Psychological)
          </span>
        </div>
      </div>
    </div>
  </article>
</div>
`;
